pkg/operator/configobservation/apiserver/observer_apiserver.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+func ObserveUserClientCABundle(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	previouslyObservedConfig := map[string]interface{}{}
+	observedConfig := map[string]interface{}{}
+
+	apiServer, err := listers.APIServerLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		glog.Warningf("apiservers.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	configMapName := apiServer.Spec.ClientCA.Name
+	if len(configMapName) == 0 {
+		return observedConfig, errs
+	}
+
+	destinationLocation := resourcesynccontroller.ResourceLocation{
+		Name:      "user-client-ca",
+		Namespace: operatorclient.OperatorNamespace,
+	}
+	sourceLocation := resourcesynccontroller.ResourceLocation{
+		Name:      apiServer.Spec.ClientCA.Name,
+		Namespace: operatorclient.GlobalUserSpecifiedConfigNamespace,
+	}
+	if err = listers.ResourceSyncer().SyncConfigMap(destinationLocation, sourceLocation); err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	return observedConfig, errs
+
+}
+
+func ObserveUserServingCertificate(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	certFilePath := []string{"servingInfo", "certFile"}
+	keyFilePath := []string{"servingInfo", "keyFile"}
+
+	previouslyObservedConfig := map[string]interface{}{}
+	if currentCertFile, _, err := unstructured.NestedString(existingConfig, certFilePath...); len(currentCertFile) > 0 || err != nil {
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if err := unstructured.SetNestedField(previouslyObservedConfig, currentCertFile, certFilePath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	if currentKeyFile, _, err := unstructured.NestedString(existingConfig, keyFilePath...); len(currentKeyFile) > 0 || err != nil {
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if err := unstructured.SetNestedField(previouslyObservedConfig, currentKeyFile, keyFilePath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	observedConfig := map[string]interface{}{}
+	apiServer, err := listers.APIServerLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		glog.Warningf("apiservers.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	servingCertSecretName := apiServer.Spec.ServingCerts.DefaultServingCertificate.Name
+
+	if len(servingCertSecretName) == 0 {
+		return observedConfig, nil
```

Comment:
you forgot to clear the previous sync rule

Reply:
Resources now sync in a `defer func()`, unless `skipResourceSync := true` to indicate that you want to keep the previous resources.